samples/java/hibernate/src/main/java/com/google/cloud/postgres/HibernateSampleTest.java

```diff
@@ -0,0 +1,260 @@
+    final Albums albums = Utils.createAlbums(singers);
+    final Venues venues = Utils.createVenue();
+    final Concerts concerts1 = Utils.createConcerts(singers, venues);
+    final Concerts concerts2 = Utils.createConcerts(singers, venues);
+    final Concerts concerts3 = Utils.createConcerts(singers, venues);
+    s.getTransaction().begin();
+    s.saveOrUpdate(singers);
+    s.saveOrUpdate(albums);
+    s.saveOrUpdate(venues);
+    s.persist(concerts1);
+    s.persist(concerts2);
+    final Tracks tracks1 = Utils.createTracks(albums.getId());
+    s.saveOrUpdate(tracks1);
+    final Tracks tracks2 = Utils.createTracks(albums.getId());
+    s.saveOrUpdate(tracks2);
+    s.persist(concerts3);
+    s.getTransaction().commit();
+    s.clear();
+
+    singersId.add(singers.getId().toString());
+    albumsId.add(albums.getId().toString());
+    venuesId.add(venues.getId().toString());
+    concertsId.add(concerts1.getId().toString());
+    concertsId.add(concerts2.getId().toString());
+    concertsId.add(concerts3.getId().toString());
+    tracksId.add(tracks1.getId().getTrackNumber());
+    tracksId.add(tracks2.getId().getTrackNumber());
+
+    logger.log(Level.INFO, "Created Singer: {0}", singers.getId());
+    logger.log(Level.INFO, "Created Albums: {0}", albums.getId());
+    logger.log(Level.INFO, "Created Venues: {0}", venues.getId());
+    logger.log(Level.INFO, "Created Concerts: {0}", concerts1.getId());
+    logger.log(Level.INFO, "Created Concerts: {0}", concerts2.getId());
+    logger.log(Level.INFO, "Created Concerts: {0}", concerts3.getId());
+    logger.log(Level.INFO, "Created Tracks: {0}", tracks1.getId());
+    logger.log(Level.INFO, "Created Tracks: {0}", tracks2.getId());
+    s.close();
+  }
+
+  public void testSessionRollback() {
+    Session s = hibernateConfiguration.openSession();
+    final Singers singers = Utils.createSingers();
+    s.getTransaction().begin();
+    s.saveOrUpdate(singers);
+    s.getTransaction().rollback();
+    s.clear();
+
+    logger.log(Level.INFO, "Singers that was saved: ", singers.getId());
+    Singers singersFromDb = s.get(Singers.class, singers.getId());
+    if (singersFromDb == null) {
+      logger.log(Level.INFO, "Singers not found as expected.");
+    } else {
+      logger.log(Level.SEVERE, "Singers found. Lookout for the error.");
+    }
+    s.close();
+  }
+
+  public void testForeignKey() {
+    Session s = hibernateConfiguration.openSession();
+
+    final Singers singers = Utils.createSingers();
+    final Albums albums = Utils.createAlbums(singers);
+
+    s.getTransaction().begin();
+    s.saveOrUpdate(singers);
+    s.persist(albums);
+    s.getTransaction().commit();
+    s.clear();
+
+    singersId.add(singers.getId().toString());
+    albumsId.add(albums.getId().toString());
+    logger.log(Level.INFO, "Created Singer: {0}", singers.getId());
+    logger.log(Level.INFO, "Created Albums: {0}", albums.getId());
+    s.close();
+  }
+
```

Comment:
Could we add a test for deleting an `Album` that has one or more `Tracks` related to it? Spanner should accept this, as `Tracks` is interleaved in `Albums` with `on delete cascade`, so I don't really expect any problems, but it would be good to know for sure that Hibernate does not somehow get confused by it.

Reply:
Done